pulp_container/app/viewsets.py

```diff
@@ -533,14 +533,44 @@ class ContainerRepositoryVersionViewSet(RepositoryVersionViewSet):
     parent_viewset = ContainerRepositoryViewSet
 
 
-class ContainerPushRepositoryViewSet(ImmutableRepositoryViewSet):
+class ContainerPushRepositoryViewSet(ReadOnlyRepositoryViewSet):
     """
     ViewSet for a container push repository.
-    """
 
+    POST and DELETE are disallowed because a push repository is tightly coupled with a
+    ContainerDistribution which handles it automatically.
+    Created - during push operation, removed - with ContainerDistribution removal
+    """
```

Comment:
Hey, i like this approach.
I think it solves https://pulp.plan.io/issues/8014 and https://pulp.plan.io/issues/7839

Reply:
@mdellweg , @ipanova , are those duplicates? They look like duplicates to me.
I'd close one as a dup, and open a new issue to cover the DELETE method.
What do you think?

Reply:
@mdellweg , @ipanova , are those duplicates? They look like duplicates to me.
I'd close one as a dup, and open a new issue to cover the DELETE method.
What do you think?

Reply:
@goosemania either way is fine. I would personally close #7839 as dup and update the title for #8014 that we are making push repos readonly - this will cover post and delete changes.

Reply:
That works as well.